### ISSUE TITLE
`Error: The AccountType property must be specified` while `azure vm create`

### ISSUE THREAD

**Issue by despairblue** [Azure/azure-xplat-cli#1486]
```bash
$ azure vm create -l 'North Europe' test1234gfys b39f27a8b8c64d52b05eac6a62ebad85__Ubuntu-14_04-LTS-amd64-server-20140724-en-us-30GB test -v
info:    Executing command vm create
warn:    --vm-size has not been specified. Defaulting to "Small".
verbose: Looking up image b39f27a8b8c64d52b05eac6a62ebad85__Ubuntu-14_04-LTS-amd64-server-20140724-en-us-30GB
Enter VM 'test' password:**********
Confirm password: **********
verbose: Checking for existence of test1234gfys cloud service
verbose: Looking up cloud service
verbose: Found existing cloud service test1234gfys
warn:    --location option will be ignored
verbose: Getting cloud service properties
verbose: Cloud service properties:
verbose: {
verbose:     uri: 'https://management.core.windows.net/44bd9beb-4124-4876-bd6d-98f1d8e39cb6/services/hostedservices/test1234gfys',
verbose:     serviceName: 'test1234gfys',
verbose:     properties: {
verbose:         extendedProperties: { ResourceGroup: 'test1234gfys', ResourceLocation: 'North Europe' },
verbose:         description: 'Implicitly created hosted service',
verbose:         location: 'North Europe',
verbose:         label: 'test1234gfys',
verbose:         status: 'Created',
verbose:         dateCreated: Wed, 17 Dec 2014 00:45:40 GMT,
verbose:         dateLastModified: Wed, 17 Dec 2014 00:45:45 GMT
verbose:     },
verbose:     statusCode: 200,
verbose:     requestId: '71328a351201504287be9d72653f9a7a'
verbose: }
verbose: Looking up deployment
verbose: Creating role
verbose: Retrieving storage accounts
verbose: Getting list of available storage accounts
verbose: Creating a new storage account 'test1234gfys141877846106'
verbose: Creating a new storage account 'test1234gfys141877846106'
error:   Unable to retrieve storage account
error:   Error: The AccountType property must be specified.
info:    Error information has been recorded to azure.err
verbose: Error: Error: The AccountType property must be specified.
    at /home/despairblue/node_modules/azure-cli/lib/commands/asm/vm/vmclient.js:3230:32
    at /home/despairblue/node_modules/azure-cli/lib/commands/asm/vm/vmclient.js:4221:18
    at /home/despairblue/node_modules/azure-cli/lib/util/blobUtils.js:78:13
    at callback_ (/home/despairblue/node_modules/azure-cli/lib/util/utils.js:262:5)
    at /home/despairblue/node_modules/azure-cli/lib/util/utils.js:285:9
    at /home/despairblue/node_modules/azure-cli/node_modules/azure/node_modules/azure-mgmt-storage/lib/storageManagementClient.js:606:16
    at /home/despairblue/node_modules/azure-cli/node_modules/azure/node_modules/azure-mgmt-storage/lib/storageManagementClient.js:421:16
    at handleRedirect (/home/despairblue/node_modules/azure-cli/lib/util/utils.js:145:9)
    at /home/despairblue/node_modules/azure-cli/lib/util/logging.js:244:7
    at handleRedirect (/home/despairblue/node_modules/azure-cli/node_modules/azure-common/lib/services/filters/redirectfilter.js:36:9)
error:   vm create command failed
```

`azure vm create --help` does not mention anything about an AccountType. The only thing I could find is a [PR](https://github.com/Azure/azure-xplat-cli/pull/1458) that supposedly made azure-cli compatible to some api change.

**Version:**
```bash
$ azure --version                                                                                                                           
0.8.13 (node: 0.10.33)
```

**Comment by jliu70** [comment]
Hi there, any idea when this might be fixed?  API changes are never fun.  The inability to launch VMs is a bit show stopper.   Thanks!!

**Comment by despairblue** [comment]
+1

**Comment by jliu70** [comment]
@despairblue @podviaznikov 
And for anyone else that is interested, a really quick and dirty hack is to checkout the latest dev branch and modify azure-xplat-cli/lib/util/utils.js

NOTE: I'm sure this is NOT the proper way to do it, but just to get up and running again, I was forced to do this ugly hack.  It's hardcoded to a specific type, but I figure that since geoReplicationEnabled is already set to false.....


azure-xplat-cli/lib/util/utils.js

  function createNewStorageAccount_ () {
    var storageAccountName = blobUtils.normalizeServiceName(name + (new Date()).getTime().toString());
+    var storageUtil = require('./storage.util');
    cli.output.verbose('Creating a new storage account \'' + storageAccountName + '\'');
    var storageOptions = {
      name: storageAccountName,
      label: storageAccountName,
+      accountType: storageUtil.AccountType.LRS,
      geoReplicationEnabled: false
    };

**Comment by amarzavery** [comment]
@despairblue @podviaznikov @jliu70 - This will be fixed in the upcoming release.

**Comment by jliu70** [comment]
@amarzavery Thank you!!

**Comment by boomkap** [comment]
When is the next release?

**Comment by amarzavery** [comment]
@boomkap Jan 23rd.

**Comment by jliu70** [comment]
@despairblue 
Just a thought - but it seems like it only attempts to create a new storage account (and thus throws the exception) if you don't already have a storage account in that location.  Maybe you can first create a storage account in that locale first.  I haven't tried to test this theory and it probably doesn't matter if you can wait until Jan 23rd when the fix will be released.

**Comment by podviaznikov** [comment]
@jliu70 I actually tried to create storage account and after that everything worked.

**Comment by amarzavery** [comment]
@jliu70 
It creates a storage account so that it can upload the image to a blob. I think if you do that step manually. i.e. create a storage account and upload the image to a blob then you can directly specify the --blob-url. This should avert the erroneous route of creating the storage account.
```
D:\sdk\xplat1\azure-xplat-cli>node bin/azure vm create -h
help:    Create a VM
help:
help:    Usage: vm create [options] <dns-name> <image> [userName] [password]
help:
help:    Options:
help:      -h, --help                             output usage information
help:      -v, --verbose                          use verbose output
help:      --json                                 use json output
help:      -u, --userName <userName>              the user name
help:      -p, --password <password>              the password
help:      -o, --community                        the <image> is a community image
help:      -c, --connect                          connect to existing VMs
help:      -l, --location <name>                  the location
help:      -a, --affinity-group <name>            the affinity group
help:      -u, --blob-url <url>                   the blob url for OS disk
help:      -z, --vm-size <size>                   the virtual machine size [small]
help:      -n, --vm-name <name>                   the virtual machine name
help:      -e, --ssh [port]                       the ssh port to enable [22]
help:      -t, --ssh-cert <pem-file|fingerprint>  the SSH certificate
help:      -P, --no-ssh-password                  indicates that the password should be removed when using --ssh-cert
help:      -r, --rdp [port]                       indicates that RDP should be enabled [3389]
help:      -w, --virtual-network-name <name>      the virtual network name
help:      -b, --subnet-names <list>              the comma-delimited subnet names
help:      -i, --public-ip <name>                 the name of the public IP address assigned to the virtual machine
help:      -S, --static-ip <ip-address>           the static IP address assigned to the virtual machine
help:      -R, --reserved-ip <name>               the name of the reserved IP address assigned to the virtual machine
help:      -A, --availability-set <name>          the name of availability set to create or use
help:      -s, --subscription <id>                the subscription id
help:      -d, --custom-data <custom-data-file>   CustomData file
```

**Comment by amarzavery** [comment]
This PR https://github.com/Azure/azure-xplat-cli/pull/1529 has been merged to the dev branch. Until January 23rd you can manually edit the file based on the changes made in the PR at the file location specified in the azure.err file

**Comment by jliu70** [comment]
@podviaznikov 
Thanks for verifying the theory.  Glad that it is a valid workaround. 

@amarzavery   Thanks for getting the fix in. Looked at your PR to learn what was the proper change.

**Comment by despairblue** [comment]
@jliu70 thank you, I was actually thinking about using the cli tools for CI but I'm not sure if that makes any sense, because of:
1. azure-xplat-cli might be rendered broken by backwards incompatible api changes made from another team at ms (an example is this bug). There seems to be no guarantee that the API will stay compatible in the foreseeable future, rendering our CI process broken until azure-xplat-cli is adapted to the api changes

2. the way issues are handled in this repo, most of them have no response or label. and I think there still might be a bug with the [password escaping](https://github.com/Azure/azure-xplat-cli/issues/1343), the ticket was closed merely by misunderstanding the bug report, not by fixing it, though I couldn't test if it was fixed by some commit without linking to the issue, because of this bug.

At the moment I don't have the time to check the other bug and evaluate the use of this in automation , does anyone use this in an automated process?

**Comment by jliu70** [comment]
@despairblue   To be honest I have only started looking at azure-xplat-cli last week after having many more difficulties with python-azure.  I quickly ran into this bug, but the overall experience I have had with azure-xplat-cli has been much more positive than with python-azure.  azure-xplat-cli seems to be the most feature complete, easy-to-use, has the most inline-usage-examples, non-Azure Powershell interface to Azure.   
Currently I'm struggling with getting a CustomScriptExtension VM extension to work with a Windows VM -- it works on the Azure Portal, but azure-xplat-cli only seems to be able to configure the extension, but not pass configuration and execute the extension.  So I tried to use Azure Powershell to execute the extension, and while it downloads the powershell script, the script runs and fails with return code of 1.  Running the the script manually is fine (as was using the extension on the Azure Portal).
I bring this up only because if Azure Powershell can't work properly, then I have to imagine that there is no "safe" tool out there.  
For now, I'll just say that Azure is slowly maturing, and hopefully things will get better and better over time.

**Comment by jliu70** [comment]
In reference to the power shell custom script extension- it does not run with a privileged account so I had to write a wrapper to use winrm to invoke command with administrative credentials.  Was confused because it worked from azure portal.

**Comment by amarzavery** [comment]
closing this issue as this will be fixed in the azure 0.8.14 release which will be available by Jan 27 2015 Tuesday
